### PR TITLE
add logs to tensorboard

### PR DESCRIPTION
**Pull request by fsx950223**
For tensorboard vis

### CODE REVIEW

File: efficientdet/keras/train_lib.py [hunk 1]
```diff
@@ -427,7 +427,8 @@ def train_step(self, data):
     images, labels = data
     with tf.GradientTape() as tape:
       cls_outputs, box_outputs = self(images, training=True)
-      det_loss, _, _, _ = self._detection_loss(cls_outputs, box_outputs, labels)
+      det_loss, cls_loss, box_loss, box_iou_loss \
```
Comment (mingxingtan) [inline efficientdet/keras/train_lib.py:430]: Prefer 

```
det_loss, cls_loss, box_loss, box_iou_loss = (
    self._detection_loss(cls_outputs, box_outputs, labels))
```

File: efficientdet/keras/train_lib.py [hunk 2]
```diff
@@ -467,7 +473,12 @@ def test_step(self, data):
     """
     images, labels = data
     cls_outputs, box_outputs = self(images, training=False)
-    det_loss, _, _, _ = self._detection_loss(cls_outputs, box_outputs, labels)
+    det_loss, cls_loss, box_loss, box_iou_loss =\
```
Comment (mingxingtan) [inline efficientdet/keras/train_lib.py:476]: Similar change (better to avoid \)

File: efficientdet/hparams_config.py [hunk 3]
```diff
@@ -249,6 +249,8 @@ def default_detection_configs():
 
   # For post-processing nms, must be a dict.
   h.nms_configs = {'method': 'hard'}
+  h.nms_score_thresh = 0.3
```
Comment (mingxingtan) [inline efficientdet/hparams_config.py:252]: How about put all these to nms_configs (postprocess.py is assuming these):

```
h.nms_configs = {
  'method': 'hard',
  'score_thresh': 0.5,
  'nms_max_output_size: 100,
}
```

I am also consider making this config file hierarchical like:

```
train:
  xxx

eval:
  yyy

nms:
  zzz
```

What do you think?

Comment (fsx950223) [reply efficientdet/hparams_config.py:252]: Good point.

**Comment by fsx950223** [comment]
Is it a bug?
https://github.com/google/automl/blob/88b9a478cee2c4d79738af79d443475b79c19d45/efficientdet/iou_utils.py#L177
pred_boxes should be [ymin, xmin, ymax, xmax] instead of box regression.
And https://github.com/google/automl/blob/de4bdd67211fd36f54c7b19ae5a2a87618ad48e2/efficientdet/dataloader.py#L271 seems wrong.

File: efficientdet/keras/train_lib.py (hunk 2 above)
Comment (mingxingtan) [reply efficientdet/keras/train_lib.py]: > Is it a bug?
> https://github.com/google/automl/blob/88b9a478cee2c4d79738af79d443475b79c19d45/efficientdet/iou_utils.py#L177
> 
> 
> pred_boxes should be [ymin, xmin, ymax, xmax] instead of box regression which is between [0,1].]]

Hmm, I haven't looked closely to iou_loss_utils.py. but it seems like " mask - tf.squeeze(_iou_per_anchor(pred_boxes, target_boxes, iou_type)))" should be " mask * (1 - tf.squeeze(_iou_per_anchor(pred_boxes, target_boxes, iou_type))))"

> And comments
> https://github.com/google/automl/blob/de4bdd67211fd36f54c7b19ae5a2a87618ad48e2/efficientdet/dataloader.py#L271
> 
> seems wrong.

Not sure about this.
